### ISSUE TITLE
Support for node pool surge upgrades

### ISSUE THREAD

**Issue by gillesdouaire** [terraform-providers/terraform-provider-google#4841]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Google has introduced a new feature for nodepool upgrades - it's called "surge upgrade". 

### New or Affected Resource(s)

* google_container_node_pool

### Potential Terraform Configuration

This feature is controlled by two distinct parameters, and there should be provision to support these:
* max-surge-upgrade
* max-unavailable-upgrade

### References

Feature is described here: https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades#surge

**Comment by Stono** [comment]
Yes please!

**Comment by Stono** [comment]
This is already available on instance group templates:

```
      + update_policy {
          + max_surge_fixed       = 0
          + max_unavailable_fixed = 1
          + min_ready_sec         = 0
          + minimal_action        = "REPLACE"
          + type                  = "PROACTIVE"
        }
```

So probably best to follow the same structure?

**Comment by chrissng** [comment]
@Stono the `google_container_node_pool` underlying GKE Beta API specifies only two settings: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/UpgradeSettings

PR addressing this issue is here: https://github.com/GoogleCloudPlatform/magic-modules/pull/2669

**Comment by danawillow** [comment]
https://github.com/GoogleCloudPlatform/magic-modules/pull/2669 was merged today and should be available in 3.1.0.
